README-dev.md

```diff
@@ -0,0 +1,266 @@
+
+This method may help in cases where the REPL is crashing inexplicably.
+Consider this the configuration of last resort.
+
+In the Alive-lsp directory:
+* Start the server in an SBCL REPL using the same port number:
+```
+        sbcl --eval "(asdf:load-system :alive-lsp)" \
+             --eval "(alive/server:start :port 8006)"
+```
+
+In the Alive directory:
+* Set the `alive.lsp.remote.host` to `"127.0.0.1"`.
+* Set the `alive.lsp.remote.port` to some number (e.g. `8006`).
+* Use the **Developer: Reload Window** command to restart the extension.
+* Exercise the server via the Alive extension to invoke the broken server code.
+  Some of this (e.g. acquiring defined packages)
+  may happen automatically at startup.
+
+In the Alive-lsp directory:
+* The REPL will crash and complain
+  `The current thread is not at the foreground`.
+* Execute `(sb-thread:release-foreground)`.
+  The REPL prompt may not be visible, just enter the command.
+* Use the Common Lisp
+  [interactive debugger](https://lispcookbook.github.io/cl-cookbook/debugging.html#the-interactive-debugger).
+
+The REPL can be run in a shell window from the Alive-lsp directory or in the
+`Terminal` view at the bottom of the VSCode workbench.
+Executing in a shell window may be marginally less confusing.
+
+With this configuration it is possible to add `(break)` calls in the code to cause a break in the REPL.
+Note that breaks in some areas will prevent the Alive extension from completing startup.
+For example, during startup Alive will request all known packages, ASDF systems, and threads.
+Breaking within the function `handle-list-pkgs` will halt processing of the packages request and
+the Alive extension startup will be delayed and (maybe) eventually fail.
+The **Locking Up the Alive Extension** section below addresses recovering from this sort of problem
+without resorting to `vim`.
+
+#### Debugging
+
+Debugging Lisp code, even with Alive, is different from most other languages.
+Breakpoints can not be marked in the editor window and stepped through in VSCode.
+The Common Lisp
+[interactive debugger](https://lispcookbook.github.io/cl-cookbook/debugging.html#the-interactive-debugger)
+is not directly available in VSCode.
+For the most part the developer is reliant on log statements which print to the `Output` view.
+
+#### Locking Up the Alive Extension
+
+If the LSP server crashes badly in the wrong place it can prevent the Alive extension from starting.
+This will in turn block some VSCode behavior, notably editing and saving files to fix the issue.
+In this case:
+* Disable the Alive extension and execute the **Developer: Reload Window** command.
+* Edit and save files or use `git` to return them to a previous working state.
+* Re-enable the Alive extension and execute the **Developer: Reload Window** command again.
+
+Another way to fix this is to
+* Remove the `alive.lsp.install.path` entry from `.vscode/settings.json`
+  using an external editor such as `vim`.
```

Comment:
There's no reason to rely on an external editor. It can all be done within VSCode.

Reply:
Removed this section entirely.